libsolidity/interface/StandardCompiler.cpp

```diff
@@ -74,22 +74,22 @@ Json::Value formatError(
 
 Json::Value formatFatalError(string const& _type, string const& _message)
 {
-	Json::Value output = Json::objectValue;
+	Json::Value output{Json::objectValue};
 	output["errors"] = Json::arrayValue;
 	output["errors"].append(formatError(Error::Severity::Error, _type, "general", _message));
 	return output;
 }
 
 Json::Value formatSourceLocation(SourceLocation const* location)
 {
-	Json::Value sourceLocation;
-	if (location && location->sourceName)
-	{
-		sourceLocation["file"] = *location->sourceName;
-		sourceLocation["start"] = location->start;
-		sourceLocation["end"] = location->end;
-	}
+	if (!(location && location->sourceName))
+		return {};
```

Comment:
This way the function is clearer that we don't want to have any value.

Reply:
Also this commit should be squashed.

Reply:
Can you make it return null explicitly?

Reply:
This last commit does that?

Reply:
Do you mean something like `Json::nullValue`? That can be done, I just copied the logic from `formatSecondarySourceLocation`, but will update both then.

Reply:
For me, it was not obvious what the default value of Json::Value is.

Reply:
Fixed.